voter-dapp/src/ActiveRequests.js

```diff
@@ -7,19 +7,69 @@ import TableHead from "@material-ui/core/TableHead";
 import TableRow from "@material-ui/core/TableRow";
 
 import { formatDate } from "./common/FormattingUtils.js";
+import { VotePhasesEnum } from "./common/Enums.js";
 
 function ActiveRequests() {
   const { drizzle, useCacheCall } = drizzleReactHooks.useDrizzle();
+  const { web3 } = drizzle;
+
   const pendingRequests = useCacheCall("Voting", "getPendingRequests");
-  if (!pendingRequests) {
+  const currentRoundId = useCacheCall("Voting", "getCurrentRoundId");
+  const votePhase = useCacheCall("Voting", "getVotePhase");
+  const account = drizzleReactHooks.useDrizzleState(drizzleState => ({
+    account: drizzleState.accounts[0]
+  })).account;
+  const initialFetchComplete = pendingRequests && currentRoundId && votePhase && account;
+
+  const voteStatuses = useCacheCall(["Voting"], call => {
+    if (!initialFetchComplete) {
+      return null;
+    }
+    return pendingRequests.map(request => ({
+      committedValue: call(
+        "Voting",
+        "getMessage",
+        account,
+        web3.utils.soliditySha3(request.identifier, request.time, currentRoundId)
+      ),
+      // Technically this call can be avoided if `votePhase == COMMIT`.
```

Comment:
Would avoiding this call in the commit case be as simple as adding a ternary after `hasRevealed:` below? Or am I missing something?

Reply:
I was trying that before, and it wasn't updating from "Reveal" to "Revealed" without a page reload, for reasons that I couldn't figure out.

I just retried it now, and everything works correctly. I must have been doing something else wrong.